app/controllers/AltAmendsController.scala

```diff
@@ -41,20 +41,22 @@ trait AltAmendsController extends ERSReturnBaseController with Authenticator wit
   def altActivityPage(): Action[AnyContent] = AuthorisedForAsync() {
     implicit user =>
       implicit request =>
-        showAltActivityPage()(user, request, hc)
+        cacheUtil.fetch[RequestObject](cacheUtil.ersRequestObject).flatMap { requestObject =>
+          showAltActivityPage(requestObject)(user, request, hc)
+        }
   }
 
-  def showAltActivityPage()(implicit authContext: AuthContext, request: Request[AnyRef], hc: HeaderCarrier): Future[Result] = {
+  def showAltActivityPage(requestObject: RequestObject)(implicit authContext: AuthContext, request: Request[AnyRef], hc: HeaderCarrier): Future[Result] = {
```

Comment:
Why is this being passed through when you've used cacheUtil inside this method?

Reply:
Remember that thing Kurt said about Inversion of Control and using RequestObject? Do That.